phac-nml/irida-next
# Fix group form errors

## What does this PR do and why?
Fixed displaying errors on create & edit group forms when validation fails for description & name that has already been taken. Sidebar & breadcrumbs are out of scope.
Fixes #191.

## Screenshots or screen recordings
![Screenshot from 2023-09-19 09-54-16](https://github.com/phac-nml/irida-next/assets/325703/56f05b93-4a92-42d1-a07c-3e2ede0a06e4)
![Screenshot from 2023-09-15 15-04-37](https://github.com/phac-nml/irida-next/assets/325703/dc63f1d4-f5e4-4097-958b-619be416d6bb)
![Screenshot from 2023-09-15 15-10-24](https://github.com/phac-nml/irida-next/assets/325703/c8c48c5f-7b4c-40e6-ac96-76bf37720830)
![Screenshot from 2023-09-15 15-03-44](https://github.com/phac-nml/irida-next/assets/325703/3b33c219-f009-438b-9029-022c745c757a)

## How to set up and validate locally
A. Test create group form validation.
   1. From the Groups Dashboard click `New group`
   2. Fill in `Group Name` with `a` and `Description` with a string longer than 255 characters.
   3. Click `Create group`. See that the form displays errors.
   4. Now update the `Group Name` to `Staphylococcus`.
   5. Click `Create group`. See that the form displays errors.
   6. Navigate back to Groups Dashboard and confirm that the group is not created as expected.
   
B. Test edit group form validation.
   1. From the Groups Dashboard click the `Staphylococcus` group.
   2. Fill in `Group Name` with `a` and `Description` with a string longer than 255 characters.
   3. Click `Save Changes`. See that the form displays errors.
   4. Navigate back to Groups Dashboard and confirm that the group was not updated as expected.
   
Note: Repeat tests for subgroups.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
